nixos/modules/security/pam.nix

```diff
@@ -279,6 +289,11 @@ let
           ${optionalString config.krb5.enable
               "account sufficient ${pam_krb5}/lib/security/pam_krb5.so"}
 
+          ${optionalString cfg.googleOsLoginAuth ''
+            account optional ${pkgs.google-compute-engine-oslogin}/lib/pam_oslogin_admin.so
```

Comment:
Rules borrowed from [google_oslogin_control](https://github.com/GoogleCloudPlatform/compute-image-packages/blob/master/google_compute_engine_oslogin/bin/google_oslogin_control) - somebody deeper into pam should definitely take a look on that.

Reply:
Is it possible to set this from the googleOSLogin module directly instead. Ideally the PAM module would provide the interface to allow googleOSLogin to set the right settings from the outside.

Reply:
Order is important here I think.

Getting that right is not very easy to do with the current module system. There is a TODO since https://github.com/NixOS/nixpkgs/commit/39bffdb34c1e1f344ba350041c10d0b07deeecb0 (from 2009) to provide such hooks. I see that outside the scope of this PR.

Reply:
I did dig deeper, and also checked back in https://github.com/GoogleCloudPlatform/compute-image-packages/issues/717.

As described there, we add the `oslogin_login.so` and `oslogin_admin.so` modules for the auth account management group at the end, and `oslogin_login.so` auth module at the beginning, with recommended upstream parameters.